app/src/main/java/com/example/amcc/view/MainFragment.java

```diff
@@ -0,0 +1,171 @@
+package com.example.amcc.view;
+
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.CalendarView;
+import android.widget.EditText;
+import android.widget.Spinner;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+
+import com.example.amcc.R;
+import com.example.amcc.model.CarDetails;
+import com.example.amcc.model.FuelType;
+import com.example.amcc.viewModel.ShardViewModel;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class MainFragment extends Fragment {
+
+    private static final String TAG = "PrintDate";
+    private Spinner spinner;
+    private EditText emissionEdtField, engineSizeField, avgConField, milePerYField;
+    private String citySelected, regDate;
+    SimpleDateFormat germanFormat;
+    CalendarView calendarView;
+
+    NavController navController;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_main, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        spinner = view.findViewById(R.id.spinnerCityHolderID);
+        emissionEdtField = view.findViewById(R.id.edtNumEmissionID);
+        engineSizeField = view.findViewById(R.id.edtNumEngSizeID);
+        avgConField = view.findViewById(R.id.edtNumConsumeID);
+        milePerYField = view.findViewById(R.id.edtNumMileAgeYearID);
+        calendarView = view.findViewById(R.id.calendarViewID);
```

Comment:
@Jaber-aldroubi  What does calenderView Here?

Reply:
@Shaftar yes, should be deleted